language/libra-vm/src/libra_vm.rs

```diff
@@ -347,13 +347,19 @@ impl LibraVM {
                     // Module publishing is currently restricted to the Association, so we choose to
                     // publish all modules under the address 0x0...1 for convenience.
                     // This will change to the sender's address once module publishing becomes open.
+                    // REVIEW: should we check that the address of the Module is in fact
+                    // `CORE_CODE_ADDRESS`?
                     let module_address = if self.on_chain_config()?.publishing_option.is_open() {
                         txn_data.sender()
                     } else {
                         account_config::CORE_CODE_ADDRESS
                     };
```

Comment:
IMO we don't have to, as we set the sender address to CORE if open publishing is not enabled, which will make the transaction fail as it wasn't signed by CORE.

However I found this a little bit confusing... Shall we stop tweaking the sender address and instead just reject a transaction directly?

Reply:
I think here we are past the verification step you mention, isn't it?

Reply:
@runtian-zhou comments?

Reply:
actually we are performing that after so @vgao1996 is correct that this is not vital at all.
But it's indeed odd we are changing a transaction in the hope it fails that way